os/drivers/serial/serial.c

```diff
@@ -926,6 +925,20 @@ static int uart_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 			}
 		}
 		break;
+		case TIOCTXSTATE: {
+			uint16_t *status = (uint16_t *)((uintptr_t)arg);
+
+			*status = TIOCS_NONE;
+	        if ((dev->xmit.head == dev->xmit.tail) && uart_txempty(dev)) {
+	            *status |= TIOCS_EMPTY;
+	        }
+
+	        if (uart_txready(dev)) {
+	            *status |= TIOCS_READY;
+	        }
```

Comment:
Sangam, could you please check the indentation?

Reply:
done